staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -98,5 +110,17 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 }
 
 func keyFunc(auds []string, token string) string {
-	return fmt.Sprintf("%#v|%v", auds, token)
+	// individually hash each input to prevent ambiguities, i.e. "xy" + "z" == "x" + "yz"
+	audsHash := sha256.Sum256([]byte(fmt.Sprintf("%#v", auds)))
+	tokenHash := sha256.Sum256([]byte(fmt.Sprintf("%#v", token)))
+
+	// then hash all inputs with a random prefix
+	hash := sha256.New()
+
+	// Write() on hash never fails
+	_, _ = hash.Write(randomCachePrefix)
+	_, _ = hash.Write(audsHash[:])
+	_, _ = hash.Write(tokenHash[:])
+
+	return base64.StdEncoding.EncodeToString(hash.Sum(nil))
```

Comment:
@lavalamp did suggest that I could directly cast this to `string` since we should not be printing it.  I can make that change if others agree on it.

Reply:
+1 to casting to a string.